lib/openscap_parser/group.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+module OpenscapParser
+  class Group < XmlNode
+    include OpenscapParser::Util
+
+    def id
+      @id ||= parsed_xml['id']
+    end
+
+    def title
+      @title ||= parsed_xml.at_css('title') &&
+        parsed_xml.at_css('title').text
+    end
+
+    def description
+      @description ||= newline_to_whitespace(
+        parsed_xml.at_css('description') &&
+          parsed_xml.at_css('description').text || ''
+      )
+    end
+
+    def rationale
+      @rationale ||= newline_to_whitespace(
+        parsed_xml.at_css('rationale') &&
+          parsed_xml.at_css('rationale').text || ''
+      )
+    end
+
+    def requires
+      @requires ||= parsed_xml.xpath('./requires') &&
+        parsed_xml.xpath('./requires/@idref').to_s&.split
+    end
+
+    def conflicts
+      @conflicts ||= parsed_xml.xpath('./conflicts') &&
+        parsed_xml.xpath('./conflicts/@idref').to_s
+    end
```

Comment:
I tested these out with real datastreams with `requires` and `conflicts` and I had to make a few updates. The way that I had it before, it was finding all requires/conflicts in the node even if they were in children elements but this was working well!

Reply:
As long as rubocop is not complaining about complexity, it can be like this :+1:

Reply:
Okay, thanks! Yeah, I had to add the `.to_s` because it was returning an `Nokigiri::XML:Node`..

Reply:
Is there any possible polymorphism? For example we could conflict a whole group or just a single rule?

Reply:
Oh, good catch. It can be either a Group or a Rule.. I could set this up to return a hash like `'Group' => 'xccdf_org.ssgproject.content_group_logging'` or `'Rule' => 'xccdf_org.ssgproject.content_rule_ensure_logrotate_activated'`

Reply:
@skateman I started doing this and then I was thinking maybe we could just store the conflicts and requires `idref` like we were doing before, without using the hash, and then when we need to access or use the conflicts or requires in `compliance-backend`, we could check if the id contains `content_group` or `content_rule`? This might not be a good way of checking the type but I can't think of a different way..

Reply:
I don't like it, but let's keep it so for now. We can refactor it later...